src/Http/Http.Features/src/IHttpResponseTrailersFeature.cs

```diff
@@ -1,10 +1,10 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 namespace Microsoft.AspNetCore.Http.Features
 {
     public interface IHttpResponseTrailersFeature
     {
-        IHeaderDictionary Trailers { get; set; }
+        IHeaderDictionary? Trailers { get; set; }
```

Comment:
This shouldn't be nullable.

Reply:
A custom dictionary can be set with this property, and it can be reset back to the default dictionary by setting it to null.

Reply:
@pranavkm Do you know if it is possible to have a property with a not null getter and a nullable setter?

Reply:
Setting IHttpRequestFeature.Headers to null does not reset it back to the default dictionary. Before these nullable changes, you were free to set Headers to null and that's what you'd get back. We should make this non-nullable like you did for Headers and match the behavior.

Reply:
We're discussing Trailers here.

My interpretation of the snippit below is you can set a custom Trailers collection, or set it back to null to get the default Trailers collection:

https://github.com/dotnet/aspnetcore/blob/f98959f630f916d74ccdb705bd02ab34b86fcf85/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.FeatureCollection.cs#L20-L40

Reply:
If we want to make this not nullable then there should be a null check in the setter:

```cs
set
{
    if (value == null) throw new ArgumentNullException(nameof(value));
    _userTrailers = value;
}
```